src/Sylius/Bundle/ApiBundle/Filter/InclusiveDateRangeFilter.php

```diff
@@ -0,0 +1,89 @@
+<?php
+
+/*
+ * This file is part of the Sylius package.
+ *
+ * (c) Sylius Sp. z o.o.
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Sylius\Bundle\ApiBundle\Filter;
+
+use ApiPlatform\Core\Bridge\Doctrine\Orm\Filter\ContextAwareFilterInterface;
+use ApiPlatform\Core\Bridge\Doctrine\Orm\Util\QueryNameGeneratorInterface;
+use Doctrine\ORM\QueryBuilder;
+
+/**
+ * When given a date-only "before" value to a DateFilter, allow it to include dates on that day
+ *
+ * To include a '1970-01-01 13:00', it transforms:
+ *      'before' => '1970-01-01'
+ * into:
+ *      'before' => '1970-01-01 23:59'
+ */
+final class InclusiveDateRangeFilter implements ContextAwareFilterInterface
```

Comment:
`Inclusive` has two meaning: before (lte) is already inclusive to strictly_before (lt) as well as time inclusive `23:59`

So I'm not sure about naming, but don't have better idea.

Reply:
I agree. In this edge case of adapting the DateFilter to solve the `Filtering orders by date from without time` test, naming it is quite tricky haha.

Perhaps naming it `DateWithoutTimeFilter` would make it less confusing ? As well as adding a point in the OpenAPI documentation that it behaves that way, and not like a standard ApiPlatform DateFilter.